tcp.c

```diff
@@ -460,6 +461,22 @@ tcp_connect(char *server)
 		g_sock = socket(addr->ai_family, addr->ai_socktype, addr->ai_protocol);
 		if (!(g_sock < 0))
 		{
+			switch (addr->ai_family) {
+				case AF_INET:
```

Comment:
`getnameinfo()` allows you to get a string without having to care about the family.